src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1677,13 +1681,19 @@ void uct_dc_mlx5_iface_set_ep_failed(uct_dc_mlx5_iface_t *iface,
         return;
     }
 
-    ep->flags |= UCT_DC_MLX5_EP_FLAG_ERR_HANDLER_INVOKED;
+    uct_dc_mlx5_fc_retrieve_ep(iface, ep, NULL);
     uct_rc_fc_restore_wnd(&iface->super.super, &ep->fc);
 
-    status  = uct_iface_handle_ep_err(&ib_iface->super.super,
-                                      &ep->super.super, ep_status);
-    log_lvl = uct_base_iface_failure_log_level(&ib_iface->super, status,
-                                               ep_status);
+    if (ep->flags & (UCT_DC_MLX5_EP_FLAG_ERR_HANDLER_INVOKED |
```

Comment:
if UCT_DC_MLX5_EP_FLAG_ERR_HANDLER_INVOKED, can we skip uct_dc_mlx5_fc_retrieve_ep and uct_rc_fc_restore_wnd ?